app/bundles/LeadBundle/Model/FieldModel.php

```diff
@@ -374,46 +374,54 @@ public function saveEntity($entity, $unlock = true)
             $entity->setIsListable(false);
         }
 
-        $event = $this->dispatchEvent('pre_save', $entity, $isNew);
-        $this->getRepository()->saveEntity($entity);
-        $this->dispatchEvent('post_save', $entity, $isNew, $event);
-
         $isUnique = $entity->getIsUniqueIdentifier();
 
-        if ($entity->getId()) {
-            //create the field as its own column in the leads table
-            $leadsSchema = $this->schemaHelperFactory->getSchemaHelper('column', $object);
-            if ($isNew || (!$isNew && !$leadsSchema->checkColumnExists($alias))) {
-                $schemaDefinition = self::getSchemaDefinition($alias, $entity->getType(), $isUnique);
-                $leadsSchema->addColumn(
+        //create the field as its own column in the leads table
+        $leadsSchema = $this->schemaHelperFactory->getSchemaHelper('column', $object);
+        if ($isNew || (!$isNew && !$leadsSchema->checkColumnExists($alias))) {
+            $schemaDefinition = self::getSchemaDefinition($alias, $type, $isUnique);
+            $leadsSchema->addColumn(
                     $schemaDefinition
                 );
+
+            try {
                 $leadsSchema->executeChanges();
+                $isCreated = true;
+            } catch (\Exception $e) {
+                $this->logger->addWarning($e->getMessage());
+                $isCreated = false;
+                throw new \ErrorException('Too many field');
```

Comment:
Hey @dongilbert, [according to the docs](https://dev.mysql.com/doc/refman/5.7/en/innodb-restrictions.html#innodb-mximums-minimums), InnoDB has a limit at 1017 columns but this only becomes relevant if you have not yet exceeded the row **size** limit of 65,535 bytes.
[The MySQL docs give the following example](https://dev.mysql.com/doc/refman/5.7/en/column-count-limit.html):
```
mysql> CREATE TABLE t (a VARCHAR(10000), b VARCHAR(10000),
       c VARCHAR(10000), d VARCHAR(10000), e VARCHAR(10000),
       f VARCHAR(10000), g VARCHAR(6000)) ENGINE=InnoDB CHARACTER SET latin1;
ERROR 1118 (42000): Row size too large. The maximum row size for the used 
table type, not counting BLOBs, is 65535. This includes storage overhead, 
check the manual. You have to change some columns to TEXT or BLOBs 
```